### ISSUE TITLE
Covers predicate fails for MULTIPOLYGON covering a MULTIPOINT with EMPTY

### ISSUE THREAD

**Issue by cuteDen-ECNU** [libgeos/geos#988]
Hi, here is a similar issue as #979.

Consider the following statement:
```sql
SELECT ST_Covers(a1, a2)
FROM ST_GeomFromText('MULTIPOLYGON(((1 0,0 1,-1 0,0 -1, 1 0)))')
As a1, 
ST_GeomFromText('MULTIPOINT(EMPTY,(0 0)) ')
As a2;
-- actual{f}, expected{t}
```
According to the definition of ST_Cover below and the point of geometry a2 lies inside, geometry a1 covers a2.

> Returns true if every point in Geometry/Geography B lies inside
 
But Postgis doesn't consider the cover relationship which seems a bug.

Version is the latest from GitHub:
 POSTGIS="3.5.0dev 3.4.0rc1-749-g42f04a29e" [EXTENSION] PGSQL="170" GEOS="3.13.0dev-CAPI-1.18.0" PROJ="8.2.1 NETWORK_ENABLED=OFF URL_ENDPOINT=https://cdn.proj.org USER_WRITABLE_DIRECTORY=/tmp/proj DATABASE_PATH=/usr/share/proj/proj.db" LIBXML="2.9.13" (core procs from "3.5.0dev 3.4.0rc1-748-gaedf3b14f" need upgrade)

**Comment by dr-jts** [comment]
A direct test:
```
bin/geosop -a "MULTIPOLYGON(((1 0,0 1,-1 0,0 -1, 1 0)))" -b "MULTIPOINT(EMPTY,(0 0))" covers
```
Interestingly, this just fails completely in JTS.

**Comment by dr-jts** [comment]
It looks like this is a regression in `main`.  It works in 3.12,

@cuteDen-ECNU can you confirm you are testing against `main`?

**Comment by cuteDen-ECNU** [comment]
Yes, I have double-checked it, and I am sure about the version. 
But it seems the issue is from PostGIS.
Because I execute the test you give:
`bin/geosop -a "POLYGON ((1 0,0 1,-1 0,0 -1, 1 0))" -b "MULTIPOINT(EMPTY,(0 0))" covers`
The result is true, and PostGIS still returns false.

**Comment by pramsey** [comment]
Hm, on my current dev env (postgis master / geos main) I am not seeing the problem
```
=# SELECT ST_Covers(a1, a2)
FROM ST_GeomFromText('MULTIPOLYGON(((1 0,0 1,-1 0,0 -1, 1 0)))')
As a1, 
ST_GeomFromText('MULTIPOINT(EMPTY,(0 0)) ')
As a2;
NOTICE:  lwgeom_api.c [352] called with n=0 and npoints=0
 st_covers 
-----------
 t
```
Actually my geos might be a week or two out of date, I'll bump it.

**Comment by cuteDen-ECNU** [comment]
Hi, I think it could be an optimization issue. In the compile process, there is a file named `configure`. ST_Cover returns the expected result `True` if I add `--enable-debug` option to execute it. However, ST_Cover returns the unexpected result `False` when I remove `--enable-debug` :
```shell
./configure --with-pgconfig=/usr/local/pgsql/bin/pg_config \
 --with-geosconfig=/usr/local/bin/geos-config \
 --without-protobuf --without-raster 
```
To give more evidence about this optimization suspect, I have replaced each '-O2' to '-O0' in `configure` file, ST_Cover returns the expected result.

Meanwhile, I have tried compiling with `-fsanitize=address`  and `-fsanitize=undefined`. It seems Postgis isn’t compatible with sanitizer:
```
psql:script/simplify.sql:43: ERROR:  could not load library "/usr/local/pgsql/lib/postgis-3.so": /usr/local/pgsql/lib/postgis-3.so: undefined symbol: __asan_option_detect_stack_use_after_return
```
So I cannot give a more accurate location.
